packages/api/cms-api/src/dam/files/files.controller.ts

```diff
@@ -80,7 +82,9 @@ export function createFilesController({ Scope: PassedScope }: { Scope?: Type<Dam
             }
 
             const uploadedFile = await this.filesService.upload(file, { ...transformedBody, scope });
-            return Object.assign(uploadedFile, { fileUrl: await this.filesService.createFileUrl(uploadedFile) });
+            const fileUrl = await this.filesService.createFileUrl(uploadedFile, Boolean(previewDamUrls));
+
+            return { ...uploadedFile, fileUrl };
```

Comment:
This fixes a minor bug where the fileUrl wasn't actually returned. I couldn't get it to work without spreading to an object (even when awaiting outside `Object.assign()`).

Maybe it has something to do with circular references because the file references an image which references the file again and so on. Maybe mikroorm does something in the background that removes the fileUrl again.